modules/Bio/EnsEMBL/Transcript.pm

```diff
@@ -3229,6 +3213,42 @@ sub get_Gene {
   return $parent_gene;
 }
 
+=head2 biotype
 
-1;
+  Arg [1]    : Arg [1] : (optional) String - the biotype to set
+  Example    : my $biotype = $transcript->biotype;
+               my $biotype = $transcript->biotype('protin_coding');
+  Description: Returns the Biotype object of this transcript.
+               When no biotype exists, defaults to 'protein_coding'.
+  Returntype : Bio::EnsEMBL::Biotype
+  Exceptions : none
+  Caller     : general
+  Status     : Stable
+
+=cut
 
+sub biotype {
+  my ( $self, $new_value) = @_;
+
+  # was a term to set given?
+  if ( defined $new_value ) {
+    $self->{'biotype'} = $new_value;
+  }
+
+  # is there no biotype? default to 'protein_coding'
+  # this is legacy behaviour and should probably be revisited
+  if ( ! defined $self->{'biotype'}) {
+    $self->{'biotype'} = 'protein_coding';
+  }
+
+  my $biotype;
+  if( defined $self->adaptor() ) {
+    my $ba = $self->adaptor()->db()->get_BiotypeAdaptor();
+    $biotype = $ba->fetch_by_name_object_type( $self->{'biotype'}, 'transcript' );
+  }
+
+  return ( $biotype || $self->{'biotype'} );
```

Comment:
Similarly here, return the cached value before trying to consult the database. Caching is valuable when you're 500 ms from a database server.